src/actions/index.js

```diff
@@ -1,38 +1,115 @@
-let taskId = 0;
+import axios from "axios";
+import types from "src/actions/Types";
 
-export const addTask = (title, description) => ({
-  type: "ADD_TASK",
+export const getTasks = () => {
+  return async dispatch => {
+    try {
+      const res = await axios.get(`http://todo-backend-express.herokuapp.com/`);
+      if (res.status !== 200) {
+        throw Error(res.statusText);
+      }
+      dispatch(getTasksSuccess(res.data));
+    } catch (err) {
+      dispatch(getTasksFailure(err.message));
+    }
+  };
+};
+
+export const addTask = (title, description) => {
+  return async dispatch => {
+    try {
+      const res = await axios.post(
+        `http://todo-backend-express.herokuapp.com/`,
+        {
+          title
+        }
+      );
+      if (res.status !== 200) {
+        throw Error(res.statusText);
+      }
+      dispatch(addTaskSuccess(res.data));
+    } catch (err) {
+      dispatch(addTasksFailure(err.message));
+    }
+  };
+};
+
+export const addTaskSuccess = task => ({
+  type: types.ADD_TASK_SUCCESS,
   payload: {
-    id: taskId++,
-    title,
-    description
-  } 
+    task
+  }
+});
+
+export const addTasksFailure = error => ({
+  type: types.FAILURE,
+  payload: {
+    error
+  }
 });
 
 export const onChangeForm = (title, description) => ({
-  type: "CHANGE_FORM",
+  type: types.CHANGE_FORM,
   payload: {
     title,
     description
   }
 });
 
 export const resetForm = (title, description) => ({
-  type: "RESET_FORM",
+  type: types.RESET_FORM,
   payload: {
     title,
     description
   }
-})
+});
 
-export const toggleDone = id => ({
-  type: "TOGGLE_DONE",
+export const toggleDone = url => ({
+  type: types.TOGGLE_DONE,
   payload: {
-    id
-  } 
-})
+    url
+  }
+});
+
+export const clearAllDone = tasks => {
+  async function clear(x) { 
+    debugger
+    const res = await axios.delete(x.url);
+    if (res.status !== 200) {
```

Comment:
maybe you can put this also in the `try-catch`, because the axios could fail and it would broke the app. Just do like the other examples like addTodo